ui-admin/src/study/surveys/VersionSelector.tsx

```diff
@@ -1,41 +1,109 @@
+  const [versionList, setVersionList] = useState<VersionedForm[]>([])
+  const [selectedVersion, setSelectedVersion] = useState<number>()
   const [isLoading, setIsLoading] = useState(true)
+  const selectId = useId()
+
   useEffect(() => {
-    Api.getSurveyVersions(studyShortname, stableId).then(result => {
-      setVersionList(result)
+    setIsLoading(true)
+    Api.getSurveyVersions(studyEnvContext.portal.shortcode, stableId).then(result => {
+      setVersionList(result.sort((a, b) => b.version - a.version))
+      setIsLoading(false)
+    }).catch(() => {
+      Store.addNotification(failureNotification('Error loading form history'))
+      setShow(false)
       setIsLoading(false)
     })
   }, [])
 
+  function loadVersion(version: number) {
+    Api.getSurvey(studyEnvContext.portal.shortcode, stableId, version).then(result => {
+      setVisibleVersionPreviews([...visibleVersionPreviews, result])
+    })
+  }
+
+  const versionOpts = versionList.map(formVersion => ({
+    label: <span>
+            Version <strong>{ formVersion.version }</strong>
+      <span className="text-muted fst-italic ms-2">
+                ({instantToDefaultString(formVersion.createdAt)})
+      </span>
+    </span>,
+    value: formVersion.version
+  })).filter(opt => visibleVersionPreviews.every(previewedVersion => previewedVersion.version !== opt.value))
+
+  const selectedOpt = versionOpts.find(opt => opt.value === selectedVersion)
+
   return <Modal show={show} onHide={() => setShow(false)}>
     <Modal.Header closeButton>
-      <Modal.Title>Select a version</Modal.Title>
+      <Modal.Title>View a previous version</Modal.Title>
     </Modal.Header>
     <Modal.Body>
       <LoadingSpinner isLoading={isLoading}>
-        <ul>
-          { versionList.map(survey => {
-            return <li key={survey.version}>
-              <button className="btn btn-secondary" onClick={() => updateVersion(survey.version)}>
-                {survey.version} {/*<span className="detail">({survey.createdAt?.substring(0, 16)})</span>*/}
-              </button>
-            </li>
-          })}
-        </ul>
+        <p>Viewing as a preview will open a new tab in the current editor.
+        Opening in read-only mode will allow you to view the full editor in an entirely new browser tab.</p>
+        <label htmlFor={selectId}>Select version to preview</label>
+        <Select inputId={selectId} options={versionOpts} value={selectedOpt} onChange={opt =>
+          setSelectedVersion(opt?.value)}/>
       </LoadingSpinner>
     </Modal.Body>
     <Modal.Footer>
-      <button type="button" className="btn btn-secondary" onClick={() => setShow(false)}>Cancel</button>
+      <Button
+        variant="primary"
+        disabled={!selectedVersion}
+        onClick={() => {
+          if (selectedVersion) {
+            loadVersion(selectedVersion)
+          }
+          setShow(false)
+        }}
+      >
+        View preview
+      </Button>
+      <Button
```

Comment:
I think it would be more friendly to have this be a link, rather than a button, since it's taking you somewhere

```
<a href={`${studyEnvFormsPath(
          studyEnvContext.portal.shortcode,
          studyEnvContext.study.shortcode,
          studyEnvContext.currentEnv.environmentName
      )}/surveys/${stableId}/${selectedVersion}?readOnly=true`}
        className="btn btn-secondary"
        aria-disabled={!selectedVersion}
         style={{pointerEvents: selectedVersion ? undefined : 'none'}}
        onClick={() => setShow(false)}
        target="_blank"
      >
        Open read-only editor <FontAwesomeIcon icon={faArrowRightFromBracket}/>
      </a>
```